src/Serilog/Context/LogContext.cs

```diff
@@ -42,11 +49,19 @@ namespace Serilog.Context
     /// </code>
     /// </example>
     /// <remarks>The scope of the context is the current logical thread, using
-    /// <see cref="CallContext.LogicalGetData"/> (and so is
-    /// preserved across async/await calls).</remarks>
+#if ASYNCLOCAL
+    /// <seealso cref="AsyncLocal{T}"/>
+#else
+    /// <seealso cref="CallContext"/>
+#endif
+    /// (and so is preserved across async/await calls).</remarks>
     public static class LogContext
     {
+#if ASYNCLOCAL
+        static readonly AsyncLocal<ImmutableStack<ILogEventEnricher>> data = new AsyncLocal<ImmutableStack<ILogEventEnricher>>();
```

Comment:
Yup. That's what happens when you learn to rely on Resharper to catch that stuff and then a project goes and breaks Resharper on you.